### ISSUE TITLE
UI for reordering Genes

### ISSUE THREAD

**Issue by Frencil** [statgen/locuszoom#18]
**Depends on #15**
## The problem

Genes are displayed on the Genes panel in the order they come back from the gene data API. Rudimentary stacking logic makes sure genes don't overlap, but there's no context for elevating "more important" genes to the top of the data layer.
## The solution

In lieu of the API itself somehow returning genes in order of importance the most elegant and immediate solution is to allow a user to reorder genes on the panel themselves through an intuitive click+drag UI. Mousing over a gene should "highlight" it by showing its bounding region. Clicking and dragging the gene should allow it to jump vertically between "tracks" but it should _not_ be able to move horizontally.

As a gene jumps to another track any genes it overlaps with should be bumped to either of the two adjacent tracks. Logic for choosing the right direction based on the situation may take some iteration to get right. This is probably most effectively done with a recursive method for bumping genes that also adds tracks in case more tracks are needed to support the new arrangement.

This creates a tricky expected behavior where a new track may be added outside of the available draw area for the panel. Options for addressing this problem:
- Scale the height of all gene tracks proportionally to the height of the data layer. If a new gene track is added during a drag event all gene tracks are scaled slightly smaller to accommodate the new track. (this seems ideal...)
- Resize the entire instance in order to also resize the genes panel to allow for new tracks to be added without changing the height of existing tracks. (this seems weird and may be counter-intuitive to users)
## Estimate

This feature will probably take at least three weeks of development to build and test. In particular, capturing the expected behavior of genes reordering themselves will probably be pretty challenging and present a lot of edge cases.

**Comment by abought** [comment]
Closing in favor of #199.
